releasenotes/notes/curve-analysis-02a702a81e014adf.yaml

```diff
@@ -0,0 +1,71 @@
+---
+features:
+  - |
+    :class:`CurveAnalysis` delegates the core fitting functionality to the LMFIT package.
+    Analysis class author can flexibly define new analysis instance
+    with LMFIT ``Model`` objects. In addition, new analysis option ``fit_method`` is added
+    so that user can perform fit with several algorithms offered by the LMFIT minimizer.
+    See LMFIT documentation for user guide.
+  - |
+    A new curve drawer option ``figure_title`` has been added. This allows user to show
+    an arbitrary string in the output figure title. See the example code below
+    to learn how to set the option.
+    
+    .. code-block:: python
+    
+      exp = MyExperiment(...)
+      exp.analysis.drawer.set_options(figure_title="Qubit0")
+    
+    Note that ``.analysis`` should be a subclass of :class:`.BaseCurveAnalysis`.
+upgrade:
+  - |
+    A new result class :class:`.CurveFitResult` is introduced. 
+    This class stores a richer context of curve fitting overview with several extra statistics.
+    This is the minimum attributes of the LMFIT ``MinimizerResult`` with some extention.
+    Fit parameters in UFloat representation are also stored while keeping
+    the correlation information, which is accessible with the ``.ufloat_params`` property.
+    Note that the value of the first analysis result entry titled with ``@Parameters_*`` 
+    has been replaced with this data format. This entry had been just a list of fit values 
+    in Python float format with covariance matrix separately stored in ``.extra`` metadata.
+    Comparing with the conventional data, new class :class:`.CurveFitResult` provides users with 
+    a better understanding of the analysis outcome. New object has prettyprint mechanism.
+  - |
+    :class:`RoughDrag` characterization experiment has been upgraded with more flexibility.
+    This experiment combines multiple DRAG parameter scans (curves) for different unit sequence
+    reptitions. Conventionally this number is limited to three curves, however, now we can define
+    more than three curves. The corresponding fit model is dynamically generated
+    based on new fit option ``reps`` in the associated analysis :class:`DragCalAnalysis`.
+    This may sometimes provide better accuracy for estimating the DRAG ``beta`` parameter.
+  - |
+    New curve drawer option ``plot_options`` has been added. This was conventionally included
```

Comment:
I think this one should be moved to feature section and be combined with teh `fiture_title` bullet as a general upgrade to CurveAnalysis drawer options.